br/tests/br_pitr/run.sh

```diff
@@ -0,0 +1,100 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+. run_services
+CUR=$(cd `dirname $0`; pwd)
+
+# const value
+PREFIX="pitr_backup" # NOTICE: don't start with 'br' because `restart services` would remove file/directory br*.
+res_file="$TEST_DIR/sql_res.$TEST_NAME.txt"
+
+# start a new cluster
+echo "restart a services"
+restart_services
+
+# prepare the data
+echo "prepare the data"
+run_sql_file $CUR/prepare_data/delete_range.sql
+# ...
+
+# start the log backup task
+echo "start log task"
+run_br --pd $PD_ADDR log start --task-name integration_test -s "local://$TEST_DIR/$PREFIX/log"
+
+# run snapshot backup
+echo "run snapshot backup"
+run_br --pd $PD_ADDR backup full -s "local://$TEST_DIR/$PREFIX/full"
+
+# load the incremental data
+echo "load the incremental data"
+run_sql_file $CUR/incremental_data/delete_range.sql
+# ...
+
+# wait checkpoint advance
+echo "wait checkpoint advance"
+sleep 10
+current_ts=$(echo $(($(date +%s%3N) << 18)))
+echo "current ts: $current_ts"
+i=0
+while true; do
+    # extract the checkpoint ts of the log backup task. If there is some error, the checkpoint ts should be empty
+    log_backup_status=$(unset BR_LOG_TO_TERM && run_br --pd $PD_ADDR log status --task-name integration_test --json 2>/dev/null)
+    echo "log backup status: $log_backup_status"
+    checkpoint_ts=$(echo "$log_backup_status" | head -n 1 | jq 'if .[0].last_errors | length  == 0 then .[0].checkpoint else empty end')
+    echo "checkpoint ts: $checkpoint_ts"
+
+    # check whether the checkpoint ts is a number
+    if [ $checkpoint_ts -gt 0 ] 2>/dev/null; then
+        # check whether the checkpoint has advanced
+        if [ $checkpoint_ts -gt $current_ts ]; then
+            echo "the checkpoint has advanced"
+            break
+        fi
+        # the checkpoint hasn't advanced
+        echo "the checkpoint hasn't advanced"
+        i=$((i+1))
+        if [ "$i" -gt 50 ]; then
+            echo 'the checkpoint lag is too large'
+            exit 1
+        fi
+        sleep 10
+    else
+        # unknown status, maybe somewhere is wrong
+        echo "TEST: [$TEST_NAME] failed to wait checkpoint advance!"
+        exit 1
+    fi
+done
+
+# dump some info from upstream cluster
+# ...
+
+# start a new cluster
+echo "restart a services"
+restart_services
+
+# PITR restore
+echo "run pitr"
+run_br --pd $PD_ADDR restore point -s "local://$TEST_DIR/$PREFIX/log" --full-backup-storage "local://$TEST_DIR/$PREFIX/full" > $res_file 2>&1
+
+# check something in downstream cluster
+echo "check br log"
+check_contains "restore log success summary"
+# check_not_contains "rewrite delete range"
```

Comment:
Notice: After release-7.1, there are duplicate `delete-range`s when drop database whose table has partition. I would modified it in the cherry-pick to release-6.5 and fix it in the next PR.